packages/circuit-ui/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,217 @@
+  invalid?: boolean;
+  /**
+   * Triggers warning message below the Checkboxes.
+   */
+  hasWarning?: boolean;
+  /**
+   * Triggers valid message below the Checkboxes.
+   */
+  showValid?: boolean;
+  /**
+   * Visually hide the label. This should only be used in rare cases and only
+   * if the purpose of the field can be inferred from other context.
+   */
+  hideLabel?: boolean;
+}
+
+const Legend = styled('legend')<Record<string, unknown>>(typography('two'));
+
+/**
+ * A group of Checkboxes.
+ */
+export const CheckboxGroup = forwardRef(
+  (
+    {
+      options,
+      value,
+      onChange,
+      'name': customName,
+      label,
+      invalid,
+      validationHint,
+      showValid,
+      disabled,
+      hasWarning,
+      hideLabel,
+      optionalLabel,
+      required,
+      'aria-describedby': descriptionId,
+      ...props
+    }: CheckboxGroupProps,
+    ref: CheckboxGroupProps['ref'],
+  ) => {
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      process.env.NODE_ENV !== 'test' &&
+      !label
+    ) {
+      throw new AccessibilityError(
+        'CheckboxGroup',
+        'The `label` prop is missing. Pass `hideLabel` if you intend to hide the label visually.',
+      );
+    }
+    const name = customName || uniqueId('checkbox-button-group_');
+    const validationHintId = uniqueId('validation-hint_');
+    const descriptionIds = `${
+      descriptionId ? `${descriptionId} ` : ''
+    }${validationHintId}`;
+
+    // Check if any of the Checkboxes are checked by default.
+    let checkedCheckboxes: { [k: string]: boolean } = {};
+    if (value) {
+      checkedCheckboxes = value.reduce(
+        (
+          accumulator: { [k: string]: boolean },
+          checkboxValue: CheckboxProps['value'],
+        ) => {
+          if (checkboxValue) {
+            accumulator[checkboxValue.toString()] = options.some(
+              (option) => checkboxValue === option.value,
+            );
+          }
+          return accumulator;
+        },
+        {},
+      );
+    }
+
+    return (
+      <FieldWrapper
+        as="fieldset"
+        role="group"
+        aria-describedby={descriptionIds}
+        name={name}
+        // @ts-expect-error TypeScript isn't smart enough to recognize the `as` prop.
+        ref={ref}
+        disabled={disabled}
+        {...props}
+      >
+        <Legend>
+          <FieldLabelText
+            label={label}
+            hideLabel={hideLabel}
+            optionalLabel={optionalLabel}
+            required={required}
+          />
+        </Legend>
+        {options &&
+          options.map(
+            ({
+              value: checkboxValue,
+              className,
+              style,
+              label: checkboxLabel,
+              ...rest
+            }) => (
+              <div
+                key={checkboxValue && checkboxValue.toString()}
+                className={className}
+                style={style}
+              >
+                <Checkbox
```

Comment:
Why did you opt for a `div` around each `Checkbox` instead of passing the style props to the `Checkbox` directly?

Reply:
I think we can indeed that we can pass the `key`, `className` and `style` props directly to the `Checkbox`. Just did it. 
I suppose we could do the same for the [`RadioButtonGroup`](https://github.com/sumup-oss/circuit-ui/blob/206b29f784f82e13f21133ac6d4961aac2fd21fe/packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx#L156-L160) right? @connor-baer @robinmetral

Reply:
> I suppose we could do the same for the [RadioButtonGroup](https://github.com/sumup-oss/circuit-ui/blob/206b29f784f82e13f21133ac6d4961aac2fd21fe/packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx#L156-L160) right?

Unlike the `Checkbox`, the `RadioButton` doesn't wrap its input and label in a wrapper `div`. I would add a wrapper to the `RadioButton` component in the next major to align the components (related to https://github.com/sumup-oss/circuit-ui/issues/1815#issuecomment-1323647714).

Reply:
Let's open an issue for this so we don't forget about it in v7 👀

Reply:
Here ya go: https://github.com/sumup-oss/circuit-ui/issues/1875